hi-ml-cpath/src/health_cpath/preprocessing/loading.py

```diff
@@ -204,38 +214,124 @@ def __call__(self, data: Dict) -> Dict:
         return data
 
 
+class LoadMaskSubROId(MapTransform, RandomizableTransform, BaseLoadROId):
+    """Transform that loads a pathology slide cropped to a sub-region of the mask bounding box (ROI).
+    The masks are supposed to be in png format, with the background in black (0) and tissue sections in categorical
+    values (e.g., 128, 255) along the RGB channels. The sub ROI is sampled from the foreground by randomly selecting a
+    label from unique values in the mask if roi_label is None. Otherwise the sub ROI is sampled from the foreground of
+    the specified label.
+
+    Operates on dictionaries, replacing the file paths in `image_key` loaded array, in (C, H, W) format. Also adds the
+    following meta-data entries:
+    - `'location'` (tuple): top-right coordinates of the bounding box
+    - `'size'` (tuple): width and height of the bounding box
+    - `'level'` (int): chosen magnification level
```

Comment:
yes, level is not magnification, it's the "magnification level" in the pyramid.
we don't have a way of retrieving magnification from the WSI object other then reading it from the properties which depends on the format anf reader (openslide vs cucim, ...), so it might be tricky to come up with something generic for all...

Reply:
OK. Would it be helpful to have the actual magnification available?

Reply:
it's not used anywhere else in the pipeline, we always use level..